src/app/src/components/Layers/ReferenceImageLayer.js

```diff
@@ -26,53 +19,29 @@ const convertCornerFromStateFormat = corner => ({
     lng: corner[1],
 });
 
-const convertImageMetadataFromStateFormat = (id, boundary, layer) => ({
-    boundary,
-    referenceImageId: id,
-    distortion: layer._corners,
-    opacity: layer.editing._transparent ? 50 : 100,
-});
-
-export default function ReferenceImageLayer() {
-    const dispatch = useDispatch();
+export default function ReferenceImageLayer({ images }) {
     const map = useMap();
+    const boundaryId = useBoundaryId();
     const referenceImageLayers = useRef({});
-    const [postReferenceImage, { error }] = useUpdateReferenceImageMutation();
-    useEndpointToastError(error);
 
-    const images = useSelector(state => state.map.referenceImages);
+    const [updateReferenceImage, { error }] =
+        useDebouncedUpdateReferenceImageMutation(boundaryId);
+    useEndpointToastError(error);
 
     const visibleImages = useMemo(
         () =>
             Object.fromEntries(
-                Object.entries(images).filter(
-                    ([, imageInfo]) => imageInfo.visible
-                )
+                images
+                    .filter(imageInfo => imageInfo.is_visible)
+                    .map(image => [image.id, image])
             ),
         [images]
     );
 
-    const updateReferenceImageDebounced = useTrailingDebounceCallback({
-        callback: (id, boundary, layer) => {
-            postReferenceImage(
-                convertImageMetadataFromStateFormat(id, boundary, layer)
-            );
-        },
-        immediateCallback: (id, boundary, layer) => {
-            dispatch(
-                api.util.updateQueryData(
-                    'updateReferenceImage',
-                    id,
-                    convertImageMetadataFromStateFormat(id, boundary, layer)
-                )
-            );
-        },
-        interval: 3000,
-    });
-
     const createLayer = useCallback(
-        ({ url, id, boundary, corners, mode }) => {
-            const layer = new L.distortableImageOverlay(url, {
+        ({ id, corners, mode }) => {
+            // TODO use reference image url here
+            const layer = new L.distortableImageOverlay('', {
```

Comment:
I wonder if we could have the django api return a url to the reference image in S3 and use that here.

Reply:
That is the plan. The API will return a signed URL that will allow temporary read access to the S3 resource, so the front-end can request the resource from there directly.